### PR TITLE
Add licenses, make it LGPLv2.1+ contributions going forward

### PR DESCRIPTION
**Pull request by IAlibay**
This license reconciles long standing issues with the way in which the MDAnalysis package was licensed.

Going forward, until such a time as the codebase is suitably relicensed, the MDAnalysis library will be packaged under the terms of the GPLv3 or any later version.

Any new contributions will be made under the conditions of the LGPLv2.1+.

PR Checklist
------------
 - [x] Tests?
 - [ ] Docs?
 - [x] CHANGELOG updated?
 - [ ] Issue raised/referenced?

----
:books: Documentation preview :books:: https://mdanalysis--4219.org.readthedocs.build/en/4219/

### CODE REVIEW

**Comment by IAlibay** [comment]
@MDAnalysis/coredevs I'm going to ask for a PR freeze until we merge this. Once this is merged we then need to make sure that all ongoing PRs agree to license their code under the terms of the LGPLv2.1 and the developer certificate of origin.

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (RMeli) [inline .github/PULL_REQUEST_TEMPLATE.md]: Does thinking this box from the UI register as an "edit"?

File: README.rst
Comment (RMeli) [inline README.rst]:
```suggestion
licenses, including LGPLv2.1+, and GPLv2+. Please see the file LICENSE_
```

File: package/doc/sphinx/source/index.rst
Comment (RMeli) [inline package/doc/sphinx/source/index.rst]:
```suggestion
.. _GNU Public Licence, version 3 or any later version:
```

**Review by RMeli — COMMENTED** [review]
A few quick comments before dosing off. Thanks @IAlibay!

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (IAlibay) [reply .github/PULL_REQUEST_TEMPLATE.md]: Yes, you can see an example of the edit history from me ticking the test checklist in this PR.

File: README.rst
Comment (IAlibay) [reply README.rst]: Thanks I pushed these two typos as a separate commit.

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (RMeli) [reply .github/PULL_REQUEST_TEMPLATE.md]: Thanks! Was on mobile, so I didn't see.

Comment (IAlibay) [reply .github/PULL_REQUEST_TEMPLATE.md]: I should have said "you can now see" - I only ticked it after you asked 🙃

File: package/doc/sphinx/source/index.rst
Comment (lilyminium) [inline package/doc/sphinx/source/index.rst]:
```suggestion
   https://www.gnu.org/licenses/gpl-3.0.en.html
```

**Review by lilyminium — CHANGES_REQUESTED** [review]
Otherwise LGTM, thanks Irfan.

File: package/doc/sphinx/source/index.rst
Comment (IAlibay) [reply package/doc/sphinx/source/index.rst]: fixed

**Review by lilyminium — APPROVED** [review]

**Comment by IAlibay** [comment]
@MDAnalysis/coredevs this is a big operational change, can I please have at least a couple more thumbs up before this gets merged?

**Review by hmacdope — APPROVED** [review]
Looks great @IAlibay, thanks for all your hard work. Very much appreciated.

**Review by RMeli — APPROVED** [review]

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (orbeckst) [inline .github/PULL_REQUEST_TEMPLATE.md]: I am not a fan of "here" links
```suggestion
- [ ] I certify that this contribution is covered by the LGPLv2.1+ license as defined in our [LICENSE](https://github.com/MDAnalysis/mdanalysis/blob/develop/LICENSE) and adheres to the [**Developer Certificate of Origin**](https://developercertificate.org/).
```

**Review by orbeckst — CHANGES_REQUESTED** [review]
Thanks. Can you please add a CHANGELOG entry, too?

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (IAlibay) [reply .github/PULL_REQUEST_TEMPLATE.md]: Done in [308e33e](https://github.com/MDAnalysis/mdanalysis/pull/4219/commits/308e33ef48c48e4bc06f51f9a3d7678ff771efa0)

**Comment by IAlibay** [comment]
Thanks all - @orbeckst can I ask for a quick re-review and then we can go ahead and merge (that's enough approving votes).

**Review by orbeckst — APPROVED** [review]
All good from my end. Thank you!

We do need a blog post as the follow up to https://www.mdanalysis.org/2022/11/07/relicensing in order to explain what we are doing. When we have the URL we can put it into the CHANGELOG retrospectively. But I am not going to hold up the PR for a blog post.

**Comment by IAlibay** [comment]
Thanks @orbeckst - yeah I just decided to get this done so that we don't accrue any further GPLv2+ contributions. The blog post will be next but might take a bit longer.

**Pull request merged**
